File: LeetCode/Easy/1399. Count Largest Group/2024-01-01_Accepted.cpp
class Solution {
public:
  int countLargestGroup(int n) {
    vector<int> count(37, 0);
    for (int i = 1; i <= n; i++) {
      count[digits(i)]++;
    }

    int maxCount = 0;
    for (int i = 1; i < 37; i++) {
      maxCount = max(maxCount, count[i]);
    }

    int ans = 0;
    for (int i = 1; i < 37; i++) {
      if (count[i] == maxCount) {
        ans++;
      }
    }
    return ans;
  }

private:
  int digits(int n) {
    int sum = 0;
    while (n) {
      sum += n % 10;
      n /= 10;
    }
    return sum;
  }
};

File: LeetCode/Easy/1394. Find Lucky Integer in an Array/2024-01-01_Accepted.py
class Solution:
  def findLucky(self, arr: List[int]) -> int:
    count = {}
    for num in arr:
      if num not in count:
        count[num] = 0
      count[num] += 1
    ans = -1
    for num in count:
      if num == count[num]:
        ans = max(ans, num)
    return ans
